src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -131,12 +132,14 @@ public MesosCloud(
     this.mesosAgentSpecTemplates = mesosAgentSpecTemplates;
     this.frameworkName = frameworkName;
 
-    if (this.frameworkId == null) {
+    this.frameworkId = frameworkId;
+    if (StringUtils.isEmpty(this.frameworkId)) {
+      logger.info("Framework ID was null.");
```

Comment:
I would log this as "Subscribing to Mesos with a new Framework ID" and actually put this after the next line so that we can log the actual ID as well. Much more helpful for debugging.

Reply:
I think USI is doing this already. I'm removing this line.